### DISCUSSION TITLE
Support for i18n

### DISCUSSION THREAD

**Discussion by reuseman** [arthelokyo/astrowind#219] [Ideas]
Hey guys, it would be nice to have support for i18n both on the landing and the blog. For reference the doc: https://docs.astro.build/en/recipes/i18n/#overview

Thanks for the work

**Comment by prototypa** [comment]
Hi @reuseman 

We are going to move this issue to Discussion since it is not a problem with **AstroWind**, following that guide the template could be adapted to other languages.  And taking advantage of the moment we are going to update AstroWind to make it very easy to return _i18n_ compatible.  Soon more updates.  Thank you so much

**Comment by reuseman** [comment]
Thanks, because right now with the current setup using dynamic routing is not clear to me how to ensure i18n over the blog

**Comment by reuseman** [comment]
This library https://github.com/jlarmstrongiv/astro-i18n-aut should be an elegant solution to keep a neet organization

**Reply by prototypa** [reply]
Hi @reuseman 

Yesterday I was playing around with extension https://github.com/jlarmstrongiv/astro-i18n-aut a bit. I like the way it handles everything, there is no need to copy pages.

In the case of **AstroWind** I found a couple of things that should be updated. Starting with the intermediate layer between the blog UI and astro:content ([utils/blog.ts](https://github.com/onwidget/astrowind/blob/main/src/utils/blog.ts)) is very outdated. This was created in the early days of Astro and many of Astro's current features didn't exist (including astro:content which was added to it later). Another thing that could represent an advantage for i18n could be to have the page data also inside astro:content.

For now I am getting some errors. I'm back today going through the code again and playing around with it a bit more. I'll send you feedback later. Thank you.

**Reply by reuseman** [reply]
I am playing a bit with blog.ts as well as dynamic routing to have everything working properly together with astro-i18n-aut. I might even open a pull request if everything works nicely

**Reply by prototypa** [reply]
Hi @reuseman 

This sounds great. I didn't have much time yesterday, I stopped at the point where I didn't know how to pass `locale` to blogs because of the way _getStaticPaths_ works (and for now AstroWind is just for static websites).

This might take a lot of time on your part, but is there a way to update a PR with the necessary changes to accommodate the [astro-18n-aut](https://github.com/jlarmstrongiv/astro-i18n-aut ) library, but without installing it? Because most people will not require its use, but will leave the doors open with the information in the `README`.

The detailed information in the `README` could be done by me if you explain it to me in this short discussion.

Otherwise I would still like to see the code to make the changes myself and leave **AstroWind** ready for i18n.

Thank you very much again.

**Comment by leonkosak** [comment]
Hi @prototypa,
https://github.com/onwidget/astrowind/issues/236
Would i18n branch be merged into main or you will maintain separated two branches in the future?

**Reply by prototypa** [reply]
Hi @leonkosak 

For now they will be in separate branches. First we have to finish a refactoring that we are developing in the main branch. **AstroWind** began to be developed at the beginning of the _Astro Framework_ and much of what _Astro_ has now did not exist, so there are some outdated ones. Also take advantage of the context to update to the new features that will be in _Astro v3_.

On the other hand, we have to mature the i18n branch. On one hand, it should be very simple to activate i18n. It would also be nice if the dependency we use for i18n matures a bit more: https://github.com/jlarmstrongiv/astro-i18n-aut

For now we will keep updates on both branches while the necessary changes are executed.

**Reply by leonkosak** [reply]
Thank you @prototypa! 👍
I noticed bigger changes in commits in August.
Do you maybe have aproximate roadmap (dates) for refactoring, i18n?
